### PR TITLE
#859 Stocktake creation

### LINKED ISSUE #859: Stocktakes: Hook up Create stocktake button
Need to hook the create stocktake button back up

### PR DESCRIPTION
**Pull request by joshxg**
Fixes #859 

- Hooks up the create stocktake button again
- Adds navigation through `stocktakeNumber` param
- Adds editable pack size in stocktake line edit

### CODE REVIEW

File: packages/inventory/src/Stocktake/DetailView/ContentArea/ContentArea.tsx [hunk 1]
```diff
@@ -33,8 +32,6 @@ export const ContentArea: FC<{
   const { rows, onChangeSortBy, sortBy } = useStocktakeRows(isGrouped);
   const columns = useStocktakeColumns({ onChangeSortBy, sortBy });
 
-  if (!rows) return <DataTableSkeleton hasGroupBy={true} />;
```
Comment (joshxg) [inline packages/inventory/src/Stocktake/DetailView/ContentArea/ContentArea.tsx:36]: This was making the skeleton show up for brand new stocktakes after loading

File: packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEditTables.tsx [hunk 2]
```diff
@@ -33,6 +33,7 @@ export const BatchTable: FC<StocktakeLineEditTableProps> = ({
       key: 'packSize',
       label: 'label.pack-size',
       width: 100,
+      Cell: NumberInputCell,
```
Comment (joshxg) [inline packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/StocktakeLineEditTables.tsx:36]: editable pack size

File: packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/hooks.ts [hunk 3]
```diff
@@ -49,6 +50,7 @@ const createDraftLine = (
     itemId: item.id,
     sellPricePerPack: 0,
     costPricePerPack: 0,
+    packSize: 1,
```
Comment (joshxg) [inline packages/inventory/src/Stocktake/DetailView/modal/StocktakeLineEdit/hooks.ts:53]: defaulting a pack size

File: packages/inventory/src/Stocktake/api/api.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
-import { DraftStocktakeLine } from './../DetailView/modal/StocktakeLineEdit/hooks';
 import {
+  generateUUID,
```
Comment (joshxg) [inline packages/inventory/src/Stocktake/api/api.ts:2]: little bit of reorganising.. think it's getting to a nice place..?

File: packages/inventory/src/Stocktake/api/hooks.ts [hunk 5]
```diff
@@ -28,29 +29,29 @@ import {
 
 export const useStocktakeApi = (): StocktakeApi => {
   const { client } = useOmSupplyApi();
-  return getSdk(client);
+  const { store } = useHostContext();
+  const queries = getStocktakeQueries(getSdk(client), store.id);
+  return { ...queries, storeId: store.id };
```
Comment (joshxg) [inline packages/inventory/src/Stocktake/api/hooks.ts:34]: I think this is nicer. Though, I would also like to put query keys here to reduce possible mistakes through the duplication we currently have.. something like `detailKey`, `listKey` 🤔

File: packages/common/src/utils/formatters.ts [hunk 6]
```diff
@@ -9,6 +9,11 @@ export const formatNaiveDate = (date?: Date | null): string | null => {
   else return null;
 };
 
+export const formatNaiveDateTime = (date?: Date | null): string | null => {
```
Comment (mark-prins) [inline packages/common/src/utils/formatters.ts:12]: ah.. NaiveDateTime has time but not timezone.. I see now. Although I thought that timezone was important 🤔 
not a real comment, just saying stuff out loud. I understand that the NaiveDateTime is going away.. so nothing to do here

File: packages/inventory/src/Stocktake/DetailView/ContentArea/ContentArea.tsx (hunk 1 above)
Comment (mark-prins) [reply packages/inventory/src/Stocktake/DetailView/ContentArea/ContentArea.tsx:36]: 🤦

File: packages/inventory/src/Stocktake/api/api.ts (hunk 4 above)
Comment (mark-prins) [reply packages/inventory/src/Stocktake/api/api.ts:2]: sure is.. looking very nice

File: packages/inventory/src/Stocktake/api/hooks.ts (hunk 5 above)
Comment (mark-prins) [reply packages/inventory/src/Stocktake/api/hooks.ts:34]: oh, that is good.

**Review by mark-prins — APPROVED** [review]
👍 LGTM!

I had something funky with the API.. not sure if you've seen this one? I already had a stocktake with number `1` and created a new stocktake.. that was assigned `1` as well so when navigating, it took me to the first stocktake, not the new one. then I had this in the list:

Screen Shot 2022-02-22 at 9 13 35 PM

there's now two stocktakes in the kawakawa store with the same number 🤷

**Comment by mark-prins** [comment]
Hm. I think that it was just the `[number]` table not updated when that first stocktake was added. fairly sure it was added manually.. so my bad.

**Comment by joshxg** [comment]
> I had something funky with the API.. not sure if you've seen this one? I already had a stocktake with number 1 and created a new stocktake.. that was assigned 1 as well so when navigating, it took me to the first stocktake, not the new one. then I had this in the list:

a composite pk on that table of storeID and number might be a good idea?

**Pull request merged**